plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/PropertiesSecretSource.java

```diff
@@ -0,0 +1,56 @@
+package io.jenkins.plugins.casc.impl.secrets;
+
+import hudson.Extension;
+import io.jenkins.plugins.casc.SecretSource;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This {@link SecretSource} implementation allows to use a .properties file for providing secrets.
+ * The default file path is {@code /run/secrets/secrets.properties}, which can be changed via
+ * {@code SECRETS} environment variable.
+ *
+ * @author <a href="mailto:d.estermann.de@gmail.com">Daniel Estermann</a>
+ */
+@Extension
```

Comment:
If you do not want this class to be used as API, you could put `@Restricted(NoExternalUse.class)` here